### DISCUSSION TITLE
Collectionview SelectionMode Multiple

### DISCUSSION THREAD

**Discussion by FM1973** [dotnet/maui#11089] [Q&A] [answered]
Hey there.

Maybe I´m too stupid, but I can´t get a collectionview to work with multiple selection the way I thought it would work.

I got a viewmodel with an oberservable collection of items. Then I got another one (same type) of selected items (still observable collection). I then configure the collectionview like this:

`
<CollectionView Grid.Row="1" ItemsSource="{Binding Dogs}" SelectedItems="{Binding SelectedDogs, Mode=TwoWay}" SelectionMode="Multiple" HorizontalOptions="StartAndExpand">`

What I expected to happen is, that when a user selects a "dog" from the list, it would be added to SelectedDogs. The same for clicking again - the dog should be removed. But the SelectedDog List remains empty.

I got arond this by using the SelectionChanged event and adding and removing dogs in my viewmmodel from the views code behind, bu thats something I really would like to avoid.

Am I missing something?

**Comment by Vroomer** [comment, marked as answer]
Hello,

make sure that SelectedDogs are of type object. You will have to cast those items in order to get the dog type.

https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/collectionview/selection

**Reply by FM1973** [reply]
@Vroomer Heck! :) That really works. Thanks a lot!

**Comment by FM1973** [comment]
So the solution of @Vroomer works on android...
When I try to do the same on IOS I get an exception:
**System.InvalidCastException:** 'Object must implement IConvertible.'
Any ideas?

**Reply by FM1973** [reply]
Never mind. I found the cause of this exception.
